### ISSUE TITLE
MM_CONFIG is ignored when storing the configuration in the database

### ISSUE THREAD

**Issue by aboukhal** [mattermost/mattermost#25696]
#### Summary
`MM_CONFIG` is used to connect to the database initially. Mattermost then reads `SqlSettings.Datasource` from the `configurations` table and ignores `MM_CONFIG` from that point forward. That means two separate configurations need to be manually kept in sync when rotating passwords.

#### Steps to reproduce
Using Mattermost version 8.0.1:
1. Ensure that you've pointed to the configuration from the database, not the `config.json`.
2. Validate that `MM_CONFIG` and `SqlSettings.Datasource` are identical
3. Change the password for the database user, and adjust `MM_CONFIG` accordingly.
4. Restart Mattermost
5. Observe that the logs indicate that connecting to the database failed.
7. Manually adjust `SqlSettings.Datasource` in the `configurations` table to the new password.
8. Restart Mattermost
9. Observe, that Mattermost starts up fine now.

#### Expected behavior
There are two alternatives that feel appropriate in this situation:

1. When `MM_CONFIG` is set, `SqlSettings.Datasource` should always be ignored.
2. It should be possible to `reset` the field `SqlSettings.Datasource` without it being immediately filled from `MM_CONFIG` again: running the command `mmctl config reset SqlSettings.DataSource` claims that `Value/s reset successfully` but reading the value immediately after, shows that it has immediately returned

#### Observed behavior (that appears unintentional) 
MM_CONFIG is only used to connect to the database to read the actual database configurations. Then a new connection is started with the connection string from the database it has already connected to.

#### Possible fixes
Not sure.

**Comment by amyblais** [comment]
Hi @aboukhal, would you be open to upgrading to a supported version to see if the issue still reproduce on a recent version? We support version v8.1 and v9.x versions.

**Comment by aboukhal** [comment]
Yes, we will upgrade after the holidays.

**Comment by aboukhal** [comment]
The issue is still reproducible on version **9.4.1**

**Comment by amyblais** [comment]
This looks related to https://mattermost.atlassian.net/browse/MM-56625.

**Comment by aboukhal** [comment]
That issue (if the link is correct) is not visible to me, so I can't assess whether it's related.

**Comment by corang** [comment]
This is an issue for us as well, any kind of automated backup/restore is completely broken, why does mattermost store the db connection string in the db?

**Comment by amyblais** [comment]
Opened a new ticket https://mattermost.atlassian.net/browse/MM-57790.

**Comment by corang** [comment]
@amyblais  Saw the note that it wasn't clear how to reproduce, so I'll sum up a quick repro here:

1. Create postgres RDS instance
2. Connect MM to rds instance using MM_CONFIG env var for db connection string
3. Put some data in MM
4. Shut down mattermost
5. Take snapshot of RDS instance
6. Delete RDS instance
7. Create new RDS instance from snapshot
8. Reconfigure MM_CONFIG env var to point to new RDS instance
9. Start MM - errors with can't connect to DB

Our current workaround is to manually edit the `configurations` table by replacing all old db connection strings with the new/correct one when restoring from a snapshot.

**Comment by agnivade** [comment]
cc @isacikgoz

**Comment by cwarnermm** [comment]
Thanks for the ping, @isacikgoz. I'll queue up this documentation update.

**Comment by aboukhal** [comment]
@isacikgoz good point, I will test `MM_SQLSETTINGS_DATASOURCE`. If that works, I'll be happy.

**Comment by aboukhal** [comment]
Okay, I've tried to implement `MM_SQLSETTINGS_DATASOURCE` as well and found out quickly why I had abstained from it so far. The environment Variables are populated through the Kubernetes Operator with the field `mattermostEnv`. So they are in there in the clear - which should not be the case for a database password!

The database credentials are passed in through the field `database.external.secret` which references a `Secret` resource that is hidden from most users, as it should be.

How can I configure `MM_SQLSETTINGS_DATASOURCE` without making the credentials publicly visible? It looks like it is not possible (yet).
